orttraining/orttraining/core/graph/gradient_builder_base.cc

```diff
@@ -143,9 +143,36 @@ void ComputeBroadcastBackwardAxesDynamic(const ArgDef& a,
               {a_op, b_op}));
 }
 
+void GradientBuilderBase::AddReduceSumNode(const ArgDef& input_arg_def,
+                                           const ArgDef& output_arg_def,
+                                           const ArgDef& reduce_axes_arg_def,
```

Comment:
since the `reduce_axes_arg_def` is only ever consumed when defining the ReduceSum node within this, can it be defined closer to when it is used i.e. within this function instead of passing it as an input?